client/src/features/ProjectPageV2/ProjectPageContent/CodeRepositories/repositories.utils.ts

```diff
@@ -70,8 +74,25 @@ export function detectSSHRepository(repositoryURL: string): boolean {
   return cleaned.match(gitUrlRegex) != null;
 }
 
+export function shouldInterrupt(
+  repositoryData: GetRepositoryApiResponse
+): RepositoryInterrupts {
+  const interruptAlways = !!(
+    !repositoryData?.metadata?.pull_permission &&
+    !(repositoryData?.connection?.status === "connected")
+  );
+  const interruptOwner = !!(
+    (!repositoryData?.metadata?.pull_permission &&
+      !(repositoryData?.connection?.status === "connected")) ||
+    (repositoryData?.metadata?.pull_permission &&
+      !repositoryData?.metadata?.push_permission &&
+      !(repositoryData?.connection?.status === "connected"))
+  );
+  return { interruptAlways, interruptOwner };
+}
+
 export function getRepositoryName(repositoryURL: string): string {
-  const canonicalUrlStr = `${repositoryURL.replace(/.git$/i, "")}`;
+  const canonicalUrlStr = `${repositoryURL.replace(/(?:\.git|\/)$/i, "")}`;
```

Comment:
```suggestion
  const canonicalUrlStr = `${repositoryURL.replace(/[/]$/, "").replace(/[.]git$/i, "")}`;
```

This correctly handles `repo.git`, `repo.git/` and `repo/`.